pkg/storage/storagewrappers/combinedtuplereader_test.go

```diff
@@ -0,0 +1,896 @@
+package storagewrappers
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"testing"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/mock/gomock"
+
+	"github.com/openfga/openfga/internal/mocks"
+	"github.com/openfga/openfga/pkg/storage"
+)
+
+var (
+	testTuples = map[string]*openfgav1.Tuple{
+		"user:11#member#group:1": {
+			Key: &openfgav1.TupleKey{
+				User:     "user:11",
+				Relation: "member",
+				Object:   "group:1",
+			},
+		},
+		"user:12#member#group:1": {
+			Key: &openfgav1.TupleKey{
+				User:     "user:12",
+				Relation: "member",
+				Object:   "group:1",
+			},
+		},
+		"user:13#member#group:1": {
+			Key: &openfgav1.TupleKey{
+				User:     "user:13",
+				Relation: "member",
+				Object:   "group:1",
+			},
+		},
+		"user:21#member#group:2": {
+			Key: &openfgav1.TupleKey{
+				User:     "user:21",
+				Relation: "member",
+				Object:   "group:2",
+			},
+		},
+		"user:22#member#group:2": {
+			Key: &openfgav1.TupleKey{
+				User:     "user:22",
+				Relation: "member",
+				Object:   "group:2",
+			},
+		},
+		"user:11#member#group:3": {
+			Key: &openfgav1.TupleKey{
+				User:     "user:11",
+				Relation: "member",
+				Object:   "group:3",
+			},
+		},
+		// userset tuples
+		"group:1#member#viewer#folder:backlog": {
+			Key: &openfgav1.TupleKey{
+				User:     "group:1#member",
+				Relation: "viewer",
+				Object:   "folder:backlog",
+			},
+		},
+	}
+)
+
+func makeMocks(t *testing.T) *mocks.MockRelationshipTupleReader {
+	controller := gomock.NewController(t)
+
+	mockRelationshipTupleReader := mocks.NewMockRelationshipTupleReader(controller)
+
+	return mockRelationshipTupleReader
+}
+
+func TestNewCombinedTupleReader(t *testing.T) {
+	reader := makeMocks(t)
+
+	tuples := []*openfgav1.TupleKey{
+		testTuples["user:11#member#group:1"].GetKey(),
+	}
+	assert.Equal(t, &combinedTupleReader{
+		RelationshipTupleReader: reader,
+		contextualTuples:        tuples,
+	}, NewCombinedTupleReader(reader, tuples))
+}
```

Comment:
I'm not sure what value `TestNewCombinedTupleReader` adds 🤔

Reply:
it was auto-generated. But we're calling it from other places so it should be fine to remove